### PR TITLE
[codex][claude] DEMO ONLY - DO NOT MERGE Consolidated critical branch coverage improvements

### PR DESCRIPTION
**Pull request by huang47**
## Summary

Consolidates the test-only critical branch coverage work from 11 constituent PRs into a single squashed commit on top of `main`.

## Coverage (critical suite, `pnpm test:coverage:critical`)

| Metric | main baseline | This PR |
|--------|---------------|---------|
| Statements | 69.59 | 94.25 |
| Branches | 60.81 | 90.34 |
| Functions | 67.39 | 91.06 |
| Lines | 70.90 | 95.18 |

Full run: 1071 test files passed, 14937 tests passed (8 skipped), 0 failures.

## Constituent PRs

This PR supersedes and consolidates the following coverage PRs, which can be closed in its favor:

#13362, #13364, #13365, #13366, #13367, #13369, #13371, #13372, #13373, #13374, #13387

Overlapping test files between branches (`useFeatureFlags.test.ts`, `useSettingUI.test.ts`, `SubgraphIONodeBase.test.ts`, `subgraphUtils.test.ts`, `arrange.test.ts`, `collections.test.ts`) were merged by union, keeping both sides' test cases.

## Notes

- All changes are **test-only**: no production source files are modified.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---

> [!NOTE]
> **Low Risk**
> All changes are unit/integration tests with no runtime behavior changes; risk is limited to CI time and test maintenance.
> 
> **Overview**
> This PR is **test-only** and expands Vitest coverage across shared utilities and graph/auth/billing composables—no production source files change.
> 
> New suites cover areas such as **`runWhenGlobalIdle`** (native vs timeout fallback), **`useCurrentUser`**, **`useLegacyBilling`**, **`useTerminal`**, canvas helpers (**`useFocusNode`**, **`useAbsolutePosition`**, **`useCanvasPositionConversion`**, **`useOverflowObserver`**), and graph UI composables (**`useFrameNodes`**, **`useGroupMenuOptions`**, **`useMoreOptionsMenu`**, **`useNodeCustomization`**, **`useSelectionOperations`**, **`useVueNodeLifecycle`**).
> 
> Existing tests gain branch and edge-case cases—for example **`downloadFile`** empty/whitespace URLs, **`useAuthActions`** billing and account flows, **`useBoundingBoxes`** draw/edit/background paths, **`useSelectionToolboxPosition`** visibility and Vue layout, **`contextMenuConverter`** submenu/error handling, **`useErrorClearingHooks`** cleanup and scan guards, **`useGraphNodeManager`** widget/property mapping, and image/selection/subgraph menu operations.
> 
> Together these additions target higher **critical-suite** branch coverage (per PR description) by exercising error paths, mocks, and lifecycle hooks that were previously untested.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit f6e9e087f779449385a1fa9698d9839efb175fe4. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by huang47** [comment]
@coderabbitai review

**Comment by huang47** [comment]
@coderabbitai approve

File: src/platform/settings/components/ColorPaletteMessage.test.ts [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    }
+  },
+  customPaletteIds: new Set<string>()
+}))
+
+vi.mock('pinia', async (importOriginal: () => Promise<typeof Pinia>) => {
+  const actual = await importOriginal()
+  return {
+    ...actual,
+    storeToRefs: (store: object) => store
```
Comment (DrJKL) [inline src/platform/settings/components/ColorPaletteMessage.test.ts:42]: Zoe sees this test mocking `vue-i18n` directly. Per the repo testing guidance, component tests should install the real i18n plugin/test setup instead of stubbing `useI18n`/`$t`; otherwise missing keys, interpolation mistakes, and plugin-integration regressions can sneak by unnoticed. Could we render this with the shared real i18n setup?

File: src/platform/settings/components/ExtensionPanel.test.ts [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+      return this.enabled.has(name)
+    },
+    isCoreExtension(name: string) {
+      return this.core.has(name)
+    },
+    isExtensionReadOnly(name: string) {
+      return this.readOnly.has(name)
+    }
+  }
+}))
```
Comment (DrJKL) [inline src/platform/settings/components/ExtensionPanel.test.ts:44]: Zoe sees this test mocking `vue-i18n` directly. Per the repo testing guidance, component tests should install the real i18n plugin/test setup instead of stubbing `useI18n`/`$t`; otherwise missing keys, interpolation mistakes, and plugin-integration regressions can sneak by unnoticed. Could we render this with the shared real i18n setup?

**Review by DrJKL — COMMENTED** [review]
Zoe spotted one recurring test-pattern issue below. Thanks for the big coverage push — let's keep the i18n checks tickly and real where these components depend on translations.

File: src/platform/settings/components/ServerConfigPanel.test.ts [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+  const actual = await importOriginal<typeof Pinia>()
+  return {
+    ...actual,
+    storeToRefs: (store: object) => store
+  }
+})
+
+vi.mock('vue-i18n', () => ({
+  useI18n: () => ({
+    t: (key: string, fallback?: string) => fallback ?? key
```
Comment (DrJKL) [inline src/platform/settings/components/ServerConfigPanel.test.ts]: Zoe sees this test mocking `vue-i18n` directly. Per the repo testing guidance, component tests should install the real i18n plugin/test setup instead of stubbing `useI18n`/`$t`; otherwise missing keys, interpolation mistakes, and plugin-integration regressions can sneak by unnoticed. Could we render this with the shared real i18n setup?

File: src/platform/settings/components/SettingDialog.test.ts [hunk 4]
```diff
@@ -0,0 +1,547 @@
...
+      value: Array<{
+        label: string
+        category?: string
+        settings: Array<{ id: string; name: string; sortOrder?: number }>
+      }>
+    }
+  },
+  handleSearch: vi.fn()
+}))
+
```
Comment (DrJKL) [inline src/platform/settings/components/SettingDialog.test.ts:63]: Zoe sees this test mocking `vue-i18n` directly. Per the repo testing guidance, component tests should install the real i18n plugin/test setup instead of stubbing `useI18n`/`$t`; otherwise missing keys, interpolation mistakes, and plugin-integration regressions can sneak by unnoticed. Could we render this with the shared real i18n setup?

File: src/platform/workflow/sharing/components/profile/ComfyHubCreateProfileForm.test.ts [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+
+import ComfyHubCreateProfileForm from './ComfyHubCreateProfileForm.vue'
+import type { ComponentProps } from 'vue-component-type-helpers'
+
+const mockCreateProfile = vi.hoisted(() => vi.fn())
+const mockToast = vi.hoisted(() => ({
+  add: vi.fn()
+}))
+const mockIsFileTooLarge = vi.hoisted(() => vi.fn())
+
```
Comment (DrJKL) [inline src/platform/workflow/sharing/components/profile/ComfyHubCreateProfileForm.test.ts:15]: Zoe sees this test mocking `vue-i18n` directly. Per the repo testing guidance, component tests should install the real i18n plugin/test setup instead of stubbing `useI18n`/`$t`; otherwise missing keys, interpolation mistakes, and plugin-integration regressions can sneak by unnoticed. Could we render this with the shared real i18n setup?

File: src/platform/workflow/sharing/components/publish/ComfyHubPublishNav.test.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
+import { render, screen, within } from '@testing-library/vue'
+import userEvent from '@testing-library/user-event'
+import { describe, expect, it, vi } from 'vitest'
+
+import ComfyHubPublishNav from './ComfyHubPublishNav.vue'
+
+vi.mock('vue-i18n', () => ({
```
Comment (DrJKL) [inline src/platform/workflow/sharing/components/publish/ComfyHubPublishNav.test.ts]: Zoe sees this test mocking `vue-i18n` directly. Per the repo testing guidance, component tests should install the real i18n plugin/test setup instead of stubbing `useI18n`/`$t`; otherwise missing keys, interpolation mistakes, and plugin-integration regressions can sneak by unnoticed. Could we render this with the shared real i18n setup?
